lib/services/cordova-migration-service.ts

```diff
@@ -107,8 +117,25 @@ export class CordovaMigrationService implements ICordovaMigrationService {
 			});
 
 			let supportedPlugins = this.pluginsForVersion(toVersion).wait();
-			plugins = _.filter(plugins, plugin => _.contains(supportedPlugins, plugin));
+			plugins = _.filter(plugins, plugin => {
+				if (plugin.indexOf('@') > -1) {
+					let pluginBasicInfo = this.$pluginsService.getPluginBasicInformation(plugin);
+					if (this.isPluginSupported(pluginBasicInfo.name, pluginBasicInfo.version, toVersion)) {
+						supportedPlugins.push(plugin);
+					} else if (!_.find(this.removedPluginsCache, p => plugin)) {
+						let remove = `Remove ${plugin} from all configuration`;
```

Comment:
Can you move the logic from this else if in a separate method and just call it here.